Objects/bytearrayobject.c

```diff
@@ -236,38 +231,47 @@ bytearray_resize_lock_held(PyObject *self, Py_ssize_t requested_size)
         }
         else {
             /* Major upsize; resize up to exact size */
-            alloc = size + 1;
+            alloc = size;
         }
     }
-    if (alloc > PY_SSIZE_T_MAX) {
+    // NOTE: offsetof() logic copied from PyBytesObject_SIZE in bytesobject.c
+    if (alloc > PY_SSIZE_T_MAX - (offsetof(PyBytesObject, ob_sval) + 1)) {
```

Comment:
Can you please move `PyBytesObject_SIZE` from bytesobject.c to pycore_bytesobject.h and rename it as `_PyBytesObject_SIZE`? Then add `#define PyBytesObject_SIZE _PyBytesObject_SIZE` to bytesobject.c (to avoid modifying the code).

Reply:
Moved, also added a `PyByteArray_SIZE_MAX` which generally replaces PY_SSIZE_T_MAX to make the checks more precise (and found one case which was still doing a ` + 1 ` / ` - 1` that doesn't need to anymore.